model/flow/sealing_segment.go

```diff
@@ -413,6 +385,53 @@ func (builder *SealingSegmentBuilder) addExecutionResult(result *ExecutionResult
 // Errors expected during normal operation:
 //   - InvalidSealingSegmentError if the added block would cause an invalid resulting segment
 func (builder *SealingSegmentBuilder) SealingSegment() (*SealingSegment, error) {
+
+	// at this point, go through all blocks and store any results which are referenced
+	// by blocks in the segment, but not contained within any blocks in the segment
+	missingExecutionResultMap := make(map[Identifier]struct{})
+
+	if builder.firstSeal != nil {
+		_, ok := builder.includedResults[builder.firstSeal.ResultID]
+		if !ok {
+			missingExecutionResultMap[builder.firstSeal.ResultID] = struct{}{}
+		}
+	}
+
+	for _, block := range append(builder.blocks, builder.extraBlocks...) {
```

Comment:
```suggestion
	for _, block := range append(builder.extraBlocks, builder.blocks...) {
```

In this particular case it makes no difference but we clearly state that extra blocks go before the sealing segment blocks so I would have the same order here so it's not confusing for anyone.